server/src/main/java/org/opensearch/extensions/action/ExtensionTransportAction.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions.action;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.extensions.ExtensionsOrchestrator;
+import org.opensearch.node.Node;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+/**
+ * The main proxy transport action used to proxy a transport request from extension to another extension
+ *
+ * @opensearch.internal
+ */
+public class ExtensionTransportAction extends HandledTransportAction<ExtensionActionRequest, ExtensionActionResponse> {
```

Comment:
This is really great. We are able to send request from one extension to another. Just wanted to know how did you test it @saratvemulapalli manually? Did you register actions of 2 different extensions?

Reply:
Yup.
- registered an action on extension
- hook up the Rest Handler API in the extension to make a transport request to the same action
- Expect the orchestrator to get the request, proxy it back to the action
- Extension responds to the transport action request
- Orchestrator gets the response, proxy it back to the caller
- Extension receives the response. 

I can paste the logs, but its huge :D